imgui_widgets.cpp

```diff
@@ -2900,6 +2901,41 @@ bool ImGui::InputTextMultiline(const char* label, char* buf, size_t buf_size, co
     return InputTextEx(label, buf, (int)buf_size, size, flags | ImGuiInputTextFlags_Multiline, callback, user_data);
 }
 
+bool ImGui::InputTextHinted(const char* label, const char* hint, char* buf, size_t buf_size, bool* is_user_data, ImGuiInputTextFlags flags, ImGuiInputTextCallback callback, void* user_data)
+{
+    ImGuiWindow* window = GetCurrentWindow();
+    const ImGuiID id = window->GetID(label);
+
+    if (id != GetActiveID()) {
+        if (buf_size > 0 && !*buf) {
+            unsigned int len = (unsigned int)strlen(hint) + 1;
+            IM_ASSERT(buf_size >= len);
+            memcpy(buf, hint, len);
+            *is_user_data = false;
+        }
+    } else {
+        if (!*is_user_data) {
+            ImGuiInputTextState& state = GImGui->InputTextState;
```

Comment:
I just realized I'm not checking if `state.ID == id`
Is it necessary here, or is it implied by the fact that `id == GetActiveID` ?